src/spiral_stc.cpp

```diff
@@ -237,19 +232,29 @@ namespace full_coverage_path_planner
         multiple_pass_counter--; // First point is already counted as visited
       }
 
-#ifdef DEBUG_PLOT
-      RCLCPP_INFO(rclcpp::get_logger("FullCoveragePathPlanner"), "Grid with path marked as visited is:");
-      gridNode_t SpiralStart = pathNodes.back();
-      printGrid(grid, visited, pathNodes, pathNodes.front(), pathNodes.back());
-#endif
+      gridNode_t lastNodeAstar = pathNodes.back();
 
       // Spiral fill from current position
       pathNodes = spiral(grid, pathNodes, visited);
```

Comment:
You're not using this value later. Except for making a copy. Why not use this?

Reply:
I use the value in line 245 to find the node in pathNodes (which get more nodes added on line 238) that correspond to the endpoint of the A* path so that I can separate the output of spiral().

Reply:
Ah, never mind. You are using it later. It wasn't visible in the PR.